flink-formats/flink-json/src/test/java/org/apache/flink/formats/json/canal/CanalJsonSerDeSchemaTest.java

```diff
@@ -319,4 +381,22 @@ public void close() {
             // do nothing
         }
     }
+
+    private static class SpecialCollector implements Collector<RowData> {
+
+        private final List<RowData> list = new ArrayList<>();
+
+        @Override
+        public void collect(RowData record) {
+            if (record.getRowKind().equals(RowKind.DELETE) && record.getInt(0) == 103) {
```

Comment:
The criteria is too specific, could we just throw exception directly no matter what the data is? We can only test the case that when exception occurs in chained downstream operator, the exception is correctly thrown.